cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/sql/PostgreSqlStructuredTableAdmin.java

```diff
@@ -141,6 +163,24 @@ private boolean tableExistsInternal(Connection connection, StructuredTableId tab
     }
   }
 
+  // Add new non-primary key columns to table
+  private void addColumnsInternal(StructuredTableSpecification spec) throws IOException {
+    StructuredTableId tableId = spec.getTableId();
+    try (Connection connection = dataSource.getConnection()) {
+      DatabaseMetaData metaData = connection.getMetaData();
+      List<String> columnStatements = getAddColumnStatements(tableId, metaData, spec.getNonPrimaryKeyFields());
+      try (Statement statement = connection.createStatement()) {
+        // Add columns
+        for (String columnStatement : columnStatements) {
```

Comment:
Does the statement need to be executed one by one? I thought we can execute multiple ALTER table at once

Reply:
Correct, I've updated it to be one single statement.